testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java

```diff
@@ -12348,6 +12367,18 @@ void testTimestampDiff(boolean coercionEnabled) {
             + "'DATETIME_TRUNC\\(<TIME\\(0\\)>, <INTERVAL HOUR>\\)'\\. "
             + "Supported form\\(s\\): 'DATETIME_TRUNC\\(<TIMESTAMP>, <DATETIME_INTERVAL>\\)'",
         false);
+    f.checkNull("datetime_trunc(CAST(NULL AS TIMESTAMP), second)");
```

Comment:
Are there tests for when the second operand is null? If not, that would be a worthwhile test to add. Same comment for other methods.